scio-test/src/it/scala/com/spotify/scio/benchmarks/ScioBenchmark.scala

```diff
@@ -148,74 +127,111 @@ final case class ScioBenchmarkLogger[F[_]](loggers: BenchmarkLogger[F]*) {
     loggers.map(_.log(benchmarks))
 }
 
-case class BenchmarkResult(
-  name: String,
-  elapsed: Option[Seconds],
-  buildNum: Long,
-  startTime: LocalDateTime,
-  finishTime: Option[LocalDateTime],
-  state: State,
-  extraArgs: Array[String],
-  metrics: Map[String, String],
-  scioVersion: String,
-  beamVersion: String
-)
+case class Metric(name: String, value: String)
+
+case class BenchmarkResult(timestamp: Instant,
+                           name: String,
+                           elapsed: Option[Long],
+                           buildNum: Long,
+                           gitHash: String,
+                           startTime: String,
+                           finishTime: Option[String],
+                           state: String,
+                           extraArgs: Array[String],
+                           metrics: List[Metric],
+                           scioVersion: String,
+                           beamVersion: String) {
+
+  def compareMetrics(other: BenchmarkResult): List[(String, Double, Double, Double)] = {
+    val otherMetrics = other.metrics.map(m => (m.name, m.value)).toMap
+    metrics.map { m =>
+      val prev = otherMetrics(m.name).toDouble
```

Comment:
this would break if we add a new metric right? can we return 0/inf/NaN or something if `otherMetrics` doesn't have `m.name`?

Reply:
yeah good point ... I'm not really happy with this actually. try something else.